### PR TITLE
Changes for lrpar

### PR DESCRIPTION
**Pull request by ltratt**
This is a series of not-very-related commits that are needed for lrpar (except for the pretty printer -- which isn't *needed* but was useful). In essence each commit either exposes more information from lrtable or changes the API to make it a bit more future proof. Hopefully, the commit messages for each should make clear what they're doing.

### CODE REVIEW

**Comment by ltratt** [comment]
[Ultimately everything in here is to make https://github.com/softdevteam/lrpar/commit/97d44f313340051d9d517450d2425122a19cef93 possible]

File: src/lib/stategraph.rs [hunk 1]
```diff
@@ -36,11 +36,55 @@ use StIdx;
...
+    }
+
+    /// Return the state pointed to by `sym` from `st_idx` or `None` otherwise.
+    pub fn edge(&self, st_idx: StIdx, sym: Symbol) -> Option<StIdx> {
+        self.edges.get(usize::from(st_idx))
+                  .and_then(|x| x.get(&sym))
+                  .cloned()
+    }
+
+    /// Return the edges for state `st`idx` or `None` if it doesn't exist.
```
Comment (ptersilie) [inline src/lib/stategraph.rs]: Typo `st`idx` instead of `st_idx`

Comment (ltratt) [inline src/lib/stategraph.rs]: Oops.

**Comment by ptersilie** [comment]
Rest looks good.

**Comment by ltratt** [comment]
Hmm, travis doesn't seem to be running?

**Comment by ltratt** [comment]
I *think* I've turned it on now. If you're OK with that commit, I will squash and (hopefully) we'll see Travis running as a bonus.

**Comment by ptersilie** [comment]
Yep, go ahead.

**Comment by ltratt** [comment]
OK, I think this is mergeable now (and Travis is back up and running!).

**Comment by ptersilie** [comment]
Merged.

**Pull request merged**